crypto/bio/bio_lib.c

```diff
@@ -174,124 +244,235 @@ int BIO_method_type(const BIO *b)
     return b->method->type;
 }
 
-int BIO_read(BIO *b, void *out, int outl)
+/*
+ * This is essentially the same as BIO_read_ex() except that it allows
+ * 0 or a -ve value to indicate failure (retryable or not) in the return. This
+ * is for compatibility with the old style BIO_read(), where existing code may
+ * make assumptions about the return value that it might get.
+ */
+static int bio_read_intern(BIO *b, void *data, size_t datal, size_t *read)
```

Comment:
I know @levitte explicitly asked for this format, but in some fonts, "datal" looks a lot like "data1".  "dlen" does not suffer from that issue (though probably has others of its own).  Anyway, I am not looking to start a bikeshed here, and would certainly be able to cope just fine with "datal".